pkg/controller.v1beta1/trial/trial_controller.go

```diff
@@ -179,7 +182,7 @@ func (r *ReconcileTrial) Reconcile(ctx context.Context, request reconcile.Reques
 	} else {
 		err := r.reconcileTrial(instance)
 		if err != nil {
-			if err == errMetricsNotReported {
+			if err == errMetricsNotReported || err == errReportMetricsFailed {
```

Comment:
```suggestion
			if errors.Is(err, errMetricsNotReported) || errors.Is(err, errReportMetricsFailed) {
```

We should not directly compare errors.

Reply:
Yeah, I agree with you.